Rubberduck.UnitTesting/UnitTesting/TestEngine.cs

```diff
@@ -143,88 +143,91 @@ private void RunWhileSuspended(IEnumerable<TestMethod> tests)
-                            OnTestCompleted(method, new TestResult(TestOutcome.Unknown, AssertMessages.Assert_ComException));
+                            RunInternal(typeLibWrapper, initializeMethods);
                         }
-                        continue;
-                    }
-                    foreach (var test in moduleTestMethods)
-                    {
-                        // no need to run setup/teardown for ignored tests
-                        if (test.Declaration.Annotations.Any(a => a.AnnotationType == AnnotationType.IgnoreTest))
+                        catch (COMException ex)
                         {
-                            OnTestCompleted(test, new TestResult(TestOutcome.Ignored));
+                            Logger.Error(ex,
+                                "Unexpected COM exception while initializing tests for module {0}. The module will be skipped.",
+                                qmn.Name);
+                            foreach (var method in moduleTestMethods)
+                            {
+                                OnTestCompleted(method, new TestResult(TestOutcome.Unknown, AssertMessages.Assert_ComException));
+                            }
                             continue;
                         }
-
-                        try
+                        foreach (var test in moduleTestMethods)
                         {
-                            fakes.StartTest();
+                            // no need to run setup/teardown for ignored tests
+                            if (test.Declaration.Annotations.Any(a => a.AnnotationType == AnnotationType.IgnoreTest))
+                            {
+                                OnTestCompleted(test, new TestResult(TestOutcome.Ignored));
+                                continue;
+                            }
+
                             try
                             {
-                                RunInternal(testInitialize);
+                                fakes.StartTest();
+                                try
+                                {
+                                    RunInternal(typeLibWrapper, testInitialize);
+                                }
+                                catch (Exception trace)
+                                {
+                                    OnTestCompleted(test, new TestResult(TestOutcome.Inconclusive, AssertMessages.Assert_TestInitializeFailure));
+                                    Logger.Trace(trace, "Unexpected Exception when running TestInitialize");
+                                    continue;
+                                }
+                                var result = RunTestMethod(typeLibWrapper, test);
+                                // we can trigger this event, because cleanup can fail without affecting the result
+                                OnTestCompleted(test, result);
+                                RunInternal(typeLibWrapper, testCleanup);
```

Comment:
Shouldn't this line be *after* the `catch` block on the line 208-212? We have a separate `catch` for the module-level cleanup on line 223-229. Why not the same treatment for the test-level cleanup?